File: src/algebra1_one_step_equations_medium.py
"""
algebra1 - one_step_equations (medium)
Generated: 2026-02-22T03:47:11.752397
"""

from problem_utils import *

def generate():
    # For one-step equations at 1300-1600 ELO range, we want:
    # - 1300: simple addition/subtraction or multiplication/division
    # - 1400-1500: multiplication/division with fractions or negatives
    # - 1600: division with fractions or slightly more complex coefficients
    
    difficulty_choice = randint(1300, 1600)
    
    if difficulty_choice <= 1350:
        # Simple addition/subtraction: x + a = b or x - a = b
        a_val = nonzero(-10, 10)
        solution_val = randint(-15, 15)
        
        if choice([True, False]):
            # x + a = b form
            b_val = solution_val + a_val
            equation = Eq(x + a_val, b_val)
            solution_text = steps(
                f"Subtract ${a_val}$ from both sides",
                f"$x = {b_val} - ({a_val})$",
                f"$x = {solution_val}$"
            )
        else:
            # x - a = b form
            b_val = solution_val - a_val
            equation = Eq(x - a_val, b_val)
            solution_text = steps(
                f"Add ${a_val}$ to both sides",
                f"$x = {b_val} + {a_val}$",
                f"$x = {solution_val}$"
            )
    
    elif difficulty_choice <= 1450:
        # Simple multiplication/division: ax = b or x/a = b
        a_val = nonzero(-8, 8)
        solution_val = nonzero(-12, 12)
        
        if choice([True, False]):
            # ax = b form
            b_val = a_val * solution_val
            equation = Eq(a_val * x, b_val)
            solution_text = steps(
                f"Divide both sides by ${a_val}$",
                f"$x = \\frac{{{b_val}}}{{{a_val}}}$",
                f"$x = {solution_val}$"
            )
        else:
            # x/a = b form
            b_val = solution_val // a_val if solution_val % a_val == 0 else Rational(solution_val, a_val)
            equation = Eq(x / a_val, b_val)
            solution_text = steps(
                f"Multiply both sides by ${a_val}$",
                f"$x = {latex(b_val)} \\cdot {a_val}$",
                f"$x = {solution_val}$"
            )
    
    elif difficulty_choice <= 1550:
        # Multiplication/division with fractions or negatives
        numerator = nonzero(-6, 6)
        denominator = choice([2, 3, 4, 5])
        solution_val = nonzero(-10, 10)
        
        if choice([True, False]):
            # (a/b)x = c form
            coeff = Rational(numerator, denominator)
            c_val = coeff * solution_val
            equation = Eq(coeff * x, c_val)
            solution_text = steps(
                f"Multiply both sides by $\\frac{{{denominator}}}{{{numerator}}}$",
                f"$x = {latex(c_val)} \\cdot \\frac{{{denominator}}}{{{numerator}}}$",
                f"$x = {solution_val}$"
            )
        else:
            # x/(a/b) = c form, which is (b/a)x = c
            c_val = randint(-15, 15)
            solution_val = Rational(numerator * c_val, denominator)
            equation = Eq(x / Rational(denominator, numerator), c_val)
            solution_text = steps(
                f"Multiply both sides by $\\frac{{{denominator}}}{{{numerator}}}$",
                f"$x = {c_val} \\cdot \\frac{{{denominator}}}{{{numerator}}}$",
                f"$x = {latex(solution_val)}$"
            )
    
    else:
        # 1550-1600: Fraction coefficients with fraction solutions
        num1 = nonzero(-5, 5)
        den1 = choice([2, 3, 4, 5])
        num2 = nonzero(-8, 8)
        den2 = choice([2, 3, 4, 5])
        
        coeff = Rational(num1, den1)
        rhs = Rational(num2, den2)
        solution_val = simplify(rhs / coeff)
        
        equation = Eq(coeff * x, rhs)
        solution_text = steps(
            f"Multiply both sides by $\\frac{{{den1}}}{{{num1}}}$",
            f"$x = \\frac{{{num2}}}{{{den2}}} \\cdot \\frac{{{den1}}}{{{num1}}}$",
            f"$x = {latex(solution_val)}$"
        )
    
    return problem(
        question=f"Solve for $x$: ${latex(equation)}$",
        answer=solution_val,
        difficulty=difficulty_choice,
        topic="algebra1/one_step_equations",
        solution=solution_text
    )

emit(generate())

File: src/algebra1_one_step_equations_medium.jl
# algebra1 - one_step_equations (medium)
# Generated: 2026-03-08T19:52:17.260008

include(joinpath(@__DIR__, "..", "julia", "src", "ProblemUtils.jl"))
using .ProblemUtils

@script x begin
    set_topic!("algebra1/one_step_equations")
    
    problem_type = rand(1:6)
    
    if problem_type == 1
        # ax = b style (division)
        ans = randint(-25, 25)
        a = nonzero(-12, 12)
        b = a * ans
        
        problem(
            question="Solve for \\(x\\): \$$(a)x = $(b)\$",
            answer=ans,
            difficulty=(1200, 1400),
            solution=steps(
                sol("Given", a*x ~ b),
                "Divide both sides by $(a)",
                sol("Answer", x ~ ans)
            ),
            time=60
        )
    elseif problem_type == 2
        # x + a = b style (subtraction)
        ans = randint(-30, 30)
        a = randint(-25, 25)
        b = ans + a
        
        problem(
            question="Solve for \\(x\\): \$x + $(a) = $(b)\$",
            answer=ans,
            difficulty=(1200, 1300),
            solution=steps(
                sol("Given", x + a ~ b),
                "Subtract $(a) from both sides",
                sol("Answer", x ~ ans)
            ),
            time=45
        )
    elseif problem_type == 3
        # x/a = b style (multiplication)
        ans = randint(-20, 20)
        a = nonzero(-10, 10)
        b = ans // a
        
        problem(
            question="Solve for \\(x\\): \$\\frac{x}{$(a)} = $(tex(b))\$",
            answer=ans,
            difficulty=(1300, 1500),
            solution=steps(
                sol("Given", x/a ~ b),
                "Multiply both sides by $(a)",
                sol("Answer", x ~ ans)
            ),
            time=60
        )
    elseif problem_type == 4
        # ax + b = c style (two-step)
        ans = randint(-20, 20)
        a = nonzero(-8, 8)
        b = randint(-30, 30)
        c = a * ans + b
        
        problem(
            question="Solve for \\(x\\): \$$(a)x + $(b) = $(c)\$",
            answer=ans,
            difficulty=(1400, 1600),
            solution=steps(
                sol("Given", a*x + b ~ c),
                "Subtract $(b) from both sides: \$$(a)x = $(c - b)\$",
                "Divide both sides by $(a)",
                sol("Answer", x ~ ans)
            ),
            time=90
        )
    elseif problem_type == 5
        # Fractional coefficient: (a/b)x = c
        ans = randint(-15, 15)
        a = nonzero(-7, 7)
        b = nonzero(-7, 7)
        c = (a * ans) // b
        
        problem(
            question="Solve for \\(x\\): \$\\frac{$(a)}{$(b)}x = $(tex(c))\$",
            answer=ans,
            difficulty=(1500, 1700),
            solution=steps(
                sol("Given", (a//b)*x ~ c),
                "Multiply both sides by the reciprocal \$\\frac{$(b)}{$(a)}\$",
                "Simplify: \$x = $(tex(c)) \\cdot \\frac{$(b)}{$(a)}\$",
                sol("Answer", x ~ ans)
            ),
            time=90
        )
    else
        # Word problem requiring equation setup
        ans = randint(5, 40)
        item_cost = nonzero(2, 15)
        fixed_cost = randint(10, 50)
        total = item_cost * ans + fixed_cost
        
        items = choice(["books", "shirts", "tickets", "notebooks", "pencils", "toys"])
        
        problem(
            question="A store charges \$$(fixed_cost) for shipping plus \$$(item_cost) per item. If the total cost is \$$(total), how many items were purchased?",
            answer=ans,
            difficulty=(1600, 1800),
            solution=steps(
                "Let \\(x\\) = number of items",
                sol("Equation", item_cost*x + fixed_cost ~ total),
                "Subtract $(fixed_cost) from both sides: \$$(item_cost)x = $(total - fixed_cost)\$",
                "Divide both sides by $(item_cost)",
                sol("Answer", x ~ ans)
            ),
            time=120
        )
    end
end